src/models/asyncapi.ts

```diff
@@ -1,3 +1,13 @@
 import { BaseModel } from "./base";
+import { Info } from "./info";
 
-export class AsyncAPIDocument extends BaseModel {}
+export class AsyncAPIDocument extends BaseModel {
+    version(): string {
+        return this.json("asyncapi");
+    }
+
+    info(): Info {
+        const doc = this.json("info");
+        return doc && new Info(doc);
```

Comment:
info is required object so you don't need to check that

```suggestion
        return new Info(this.json("info"));
```

Reply:
I applied this but I'm not very convinced about it right now. Shouldn't we handle this anyway for the matter of consistency and testing?

Reply:
But what do you mean? in v2 `info` is required so it cannot be undefined so `info` method should always return `Info` instance. Or maybe I don't understand something. 🤔 ]

Reply:
Remember we are building here an API based on intents. `Info` model is not a 1:1 with the `info` key (even though it is right now). 

I think it's not the fact we check if it's empty or not what it "shocks"me. It's the differences between major spec versions what and how to handle them. But this is out of this PR and will work right after merging it so sooner the better. So Nvm by now :P

Reply:
ok i understand. However, it would be necessary to have a consensus on how to do it in relation to the AsyncAPI version because it will probably be the most difficult task here.

It was a mistake that I didn't participate at all in the discussion about IntentAPI, because e.g. in react-component or markdown-template intent api won't be used at all, so I'm still thinking about 1:1 of API into specification structure.